src/pykeen/nn/functional.py

```diff
@@ -246,26 +246,13 @@ def convkb_interaction(
     assert conv.weight.shape == (num_filters, 1, 1, 3)
 
     # compute conv(stack(h, r, t))
-    # prepare input shapes for broadcasting
-    # (*batch_dims, 1, d)
-    h = h.unsqueeze(dim=-2)
-    r = r.unsqueeze(dim=-2)
-    t = t.unsqueeze(dim=-2)
-
     # conv.weight.shape = (C_out, C_in, kernel_size[0], kernel_size[1])
     # here, kernel_size = (1, 3), C_in = 1, C_out = num_filters
-    # -> conv_head, conv_rel, conv_tail shapes: (num_filters,)
-    # reshape to (..., f, 1)
-    conv_head, conv_rel, conv_tail, conv_bias = [
-        c.view(*make_ones_like(h.shape[:-2]), num_filters, 1) for c in list(conv.weight[:, 0, 0, :].t()) + [conv.bias]
-    ]
-
-    # convolve -> output.shape: (*, embedding_dim, num_filters)
-    h = conv_head @ h
-    r = conv_rel @ r
-    t = conv_tail @ t
-
-    x = tensor_sum(conv_bias, h, r, t)
+    # => (num_filters, 1, 1, 3)
+    x = tensor_sum(
```

Comment:
this is a work of art in that it fits on one line, but this needs some serious docstrings to explain what's going on

Reply:
also it looks like some of the comments from above survived, are these relevant?

Reply:
these explain what is going on 😛 I'll extend that part

Reply:
https://github.com/pykeen/pykeen/pull/978/commits/ff9c1c39ef6d19afc499bf3892653920ba9d9e58